fix(react): build errors with ReactTypescriptProject

agdimech opened the pull request:
This PR fixes 2 build issues with the ReactTypescriptProject:

1. react-app-env.d.ts was not being synthesized into the src directory. This PR ensures it is emitted into the src root whcih fixes the issue with svg imports.
2. Removed redundant React import from unit test as this was causing TSErrors for unused imports.

Fixes: #1635 
---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Chriscbr commented:
@agdimech Is there an issue describing what the original issue or build error(s) are?

agdimech commented:
> @agdimech Is there an issue describing what the original issue or build error(s) are?

Yes, #1635. I have updated the PR description and added a 'Fixes: #1635' to the latest commit msg.
